File: explainer/src/main/java/com/example/rag/explainer/Explainer.java
package com.example.rag.explainer;

import core.explanation.ExplainerSystem;
import org.springframework.stereotype.Component;

@Component
class Explainer implements ExplainerSystem {
    @Override
    public String explain(String contents, String personalData) {
        String response = contents + "를 '" + personalData + "' 바탕으로 친절한 설명해주세요";
        System.out.println(response);
        return response;
    }
}

File: explainer/src/main/kotlin/com/example/rag/explainer/Explainer.kt
package com.example.rag.explainer

import core.explanation.ExplainerSystem

class Explainer: ExplainerSystem {
    override fun explain(content: String, personalData: String?) =
        "내용: $content 를 개인정보: $personalData 를 바탕으로 친절한 설명해주세요"
}
